DotBot.py

```diff
@@ -54,24 +55,25 @@ def receive(self, frame):
         # parse frame, extract myMovement
         myMovement = frame['movements'][self.dotBotId]
 
-        # filter out duplicates
-        if myMovement['seqNumMovement']==self.lastSeqNumMovement:
+        now = self.simEngine.currentTime()
 
-            self.simEngine.schedule(self.simEngine.currentTime() + 1, self._transmit)
+        # filter out duplicates
+        if myMovement['seqNumMovement'] == self.lastSeqNumMovement:
             return
 
         self.lastSeqNumMovement   = myMovement['seqNumMovement']
 
         # if I get here I have received a NEW movement
-        
+
         # cancel notification retransmission
+        self.simEngine.cancelEvent(tag = (str(self.seqNumNotification) + str(self.dotBotId)))
```

Comment:
I don't understadn how this works. What you want to cancel are all retransmission scheduled by the DotBot inthe future. So the tag you're looking for is `"retrnansmission_DotBot_{}".format(self.dotBotId)`.

The tag you are using refers to a particular `self.seqNumNotification`, which changes over time. Why this complexity?